crates/oxc_linter/src/rules/react/jsx_props_no_spread_multi.rs

```diff
@@ -0,0 +1,100 @@
+use std::collections::HashMap;
+
+use oxc_ast::{ast::JSXAttributeItem, AstKind};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Atom, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn jsx_props_no_spread_multi_diagnostic(span0: Span, span1: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Disallow JSX prop spreading the same identifier multiple times.")
+        .with_help("Remove duplicate spread attributes.")
+        .with_labels([span0, span1])
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct JsxPropsNoSpreadMulti;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Enforces that any unique expression is only spread once.
+    /// Generally spreading the same expression twice is an indicator of a mistake since any attribute between the spreads may be overridden when the intent was not to.
+    /// Even when that is not the case this will lead to unnecessary computations being performed.
+    ///
+    /// ### Example
+    /// ```jsx
+    /// // Bad
+    /// <App {...props} myAttr="1" {...props} />
+    ///
+    /// // Good
+    /// <App myAttr="1" {...props} />
+    /// <App {...props} myAttr="1" />
+    /// ```
+    JsxPropsNoSpreadMulti,
+    correctness,
+);
+
+impl Rule for JsxPropsNoSpreadMulti {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::JSXOpeningElement(jsx_opening_el) = node.kind() {
+            let spread_attrs = jsx_opening_el.attributes.iter().filter_map(|attr| {
+                if let JSXAttributeItem::SpreadAttribute(spread_attr) = attr {
+                    if spread_attr.argument.is_identifier_reference() {
+                        return Some(spread_attr);
+                    }
+                }
+                None
+            });
+
+            let mut identifier_names: HashMap<&Atom, Span> = HashMap::new();
+
+            for spread_attr in spread_attrs {
```

Comment:
Maybe we should collect all duplicate spreads before reporting diagnostics so that
```tsx
<Foo {...props} {...props} {...props} />
```
only reports a single diagnostic?

Reply:
I’ve corrected it to report only a single diagnostic result.
7e73647ad9f4d4bf84102c400552e9521c20a290